sql/sql_planner.cc

```diff
@@ -954,7 +954,18 @@ void Optimize_table_order::best_access_path(JOIN_TAB *tab,
   table_map ref_depend_map= 0;
   uint used_key_parts= 0;
 
-  if (tab->keyuse() != NULL)
+  /*
+    The primary key of the tianmu engine does not support delete and update statements.
+    The following codes can be deleted after subsequent support
+  */
+  bool tianmu_engine = table->s->db_type() ? table->s->db_type()->db_type == DB_TYPE_TIANMU: false;
+  enum_sql_command sqlCommand = table->in_use->lex->sql_command;
+  bool tianmuDeleteOrUpdate = (tianmu_engine && (sqlCommand == SQLCOM_DELETE ||
```

Comment:
I think this logic is better in optimization. in optimization, if we foud the table is Tianmu table. we disable the `key` information. 

Then, here, tab->keyuse() of a Tianmu table is in disable status.